File: STRIVER/video/vid-5/pattern21/p21.cpp
// ****
// *  *
// *  *
// ****

// formula: 
// printstars: only when (i==0 or i==row-1 or j==0 or j==row-1)

#include <iostream>
using namespace std;

int main()
{
    int row = 5;

    for (char i = 0; i < row; i++){

        for(int j = 0; j < row; j++){
            if(i==0 || i==row-1 || j==0 || j==row-1)cout<<"*";
            else cout<<" ";
        }
        cout<<endl;
    }

    return 0;
}

File: STRIVER/video/vid-5/pattern21/p21.py
# ****
# *  *
# *  *
# ****

# formula: 
# printstars: only when (i==0 or i==row-1 or j==0 or j==row-1)

def p21(row:int)->None:

    for i in range(row):
        for j in range(row):
            if(i==0 or i==row-1 or j==0 or j==row-1):
                print("*",end="")
            else:
                print(" ",end="")
        print(" ")

row = 6
p21(row)
